resources/roca_main.brs

```diff
@@ -1,38 +1,37 @@
-sub main()
+function main() as object
     basePath = "tests"
     files = MatchFiles("pkg:/" + basePath, "*.test.brs")
 
     rootSuites = []
+    filesWithFocusedCases = []
     for each file in files
-        path = ["pkg:", basePath, file].join("/")
-        suite = _brs_.runInScope(path, {})
+        filePath = [basePath, file].join("/")
+        suite = _brs_.runInScope(["pkg:", filePath].join("/"), {})
 
         if suite = invalid then
             print "Error running tests: Runtime exception occurred in " + [basePath, file].join("/")
             return
         end if
 
-        if GetInterface(suite, "ifArray") <> invalid then
-            rootSuites.append(suite)
-        else
-            rootSuites.push(suite)
+        if GetInterface(suite, "ifArray") = invalid then
+            suite = [suite]
         end if
-    end for
 
-    numFocusedSuites = 0
-    focusedCasesDetected = false
-    for each suite in rootSuites
-        if suite.mode = "focus" or suite.__state.hasFocusedDescendants then
-            numFocusedSuites++
-            focusedCasesDetected = true
-        end if
+        for each subSuite in suite
+            if subSuite.mode = "focus" or subSuite.__state.hasFocusedDescendants then
+                filesWithFocusedCases.push(filePath)
+            end if
+
+            rootSuites.push(subSuite)
+        end for
     end for
 
     tap = tap()
     tap.version()
 
+    focusedCasesDetected = filesWithFocusedCases.count() > 0
```

Comment:
Mostly a nit, but rather than doing this boolean evaluation, here, consider just holding onto the value of `count()`. That we don't need to reevaluate it, below, on line 34:
```brightscript
numFocusedFiles = filesWithFocusedCases.count()
if numFocusedCases > 0 then
    tap.plan(numFocusedFiles)
else
    tap.plan(rootSuites.count())
end if
```

Reply:
We use `focusedCasesDetected` in 3 other places later on, and it feels a little overkill to have so many variables (thus adding more "sources of truth") when `.count()` exists.